src/flint/fq_default_poly.jl

```diff
@@ -513,6 +513,8 @@ end
 ################################################################################
 
 function is_irreducible(x::FqPolyRingElem)
+  is_zero(x) && return false
+  is_unit(x) && return false
```

Comment:
How about this (at least when the coefficients are in a field) ?
```suggestion
  is_constant(x) && return false
```